spec/controllers/documents/intro_controller_spec.rb

```diff
@@ -39,6 +39,17 @@
       end
     end
 
+    context "when no documents are required" do
+      before do
+        allow(subject.class).to receive(:show?).and_return(false)
```

Comment:
Not a blocker: it's tidier to do `described_class` rather than `subject.class`.

I encourage you to make this one change, then merge (no need for re-review if you make this change).